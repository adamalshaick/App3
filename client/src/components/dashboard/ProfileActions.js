import React from "react";
import { Link } from "react-router-dom";

const ProfileActions = () => {
  return (
    <div
      style={{ position: "absolute", top: "0", left: "20px" }}
      className="btn-group mb-4"
      role="group"
    >
      <Link to="/edit-profile" className="btn btn-dark mt-3">
        <i className="fas fa-user-circle text-light mr-1" /> Edit Profile
      </Link>
    </div>
  );
};

export default ProfileActions;
