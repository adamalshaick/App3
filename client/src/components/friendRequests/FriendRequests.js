import React, { Component } from "react";
import RequestItem from "./RequestItem";
import PropTypes from "prop-types";

class FriendRequests extends Component {
  render() {
    const { profile, profiles } = this.props;
    return profile.friendRequests.map(request => (
      <RequestItem
        key={request._id}
        request={request}
        profile={profile}
        profiles={profiles}
      />
    ));
  }
}

FriendRequests.propTypes = {
  profile: PropTypes.object.isRequired
};

export default FriendRequests;
