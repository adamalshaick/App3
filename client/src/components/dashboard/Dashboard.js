import React from "react";
import PropTypes from "prop-types";
import ProfileContent from "../profile/ProfileContent";
import FriendsFeed from "../friends/FriendsFeed";
import FriendRequests from "../friendRequests/FriendRequests";
import { Header } from "../common/styles/Header";
import fetchProfile from "../common/hoc/fetchProfile";

const Dashboard = ({ auth, profile }) => {
  return (
    <article className="container entry">
      {/* return friend requests if there are any */}
      {profile.currentProfile.friendRequests &&
      profile.currentProfile.friendRequests.length ? (
        <FriendRequests profile={profile.currentProfile} />
      ) : null}
      <div className="row mt-5">
        <section
          style={{ borderRight: "whitesmoke solid 2px" }}
          className="col-md-6"
        >
          <ProfileContent profile={profile.currentProfile} user={auth.user} />
        </section>
        <section className="col-md-6">
          <Header className="text-center">Your Friends</Header>
          <hr />
          <FriendsFeed currentProfile={profile.currentProfile} />
        </section>
      </div>
    </article>
  );
};

Dashboard.propTypes = {
  auth: PropTypes.object.isRequired,
  profile: PropTypes.object.isRequired
};

export default fetchProfile(Dashboard);
