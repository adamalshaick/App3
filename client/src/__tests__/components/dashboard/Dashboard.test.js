import React from "react";
import { shallow } from "enzyme";
import { Dashboard } from "../../../components/dashboard/Dashboard";
import toJson from "enzyme-to-json";
import FriendRequests from "../../../components/friendRequests/FriendRequests";

describe("render component", () => {
  const Props = {
    auth: {},
    profile: {
      currentProfile: {
        friendRequests: []
      }
    }
  };

  const wrapper = shallow(<Dashboard {...Props} />);
  it("renders component", () => {
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});

describe("no friend requests", () => {
  const Props = {
    auth: {},
    profile: {
      currentProfile: {
        friendRequests: []
      }
    }
  };

  const wrapper = shallow(<Dashboard {...Props} />);
  it("doesnt display friend requests array", () => {
    expect(wrapper.find(FriendRequests).length).toEqual(0);
  });
});

describe("there are friend requests", () => {
  const PropsWithRequests = {
    auth: {},
    profile: {
      currentProfile: {
        friendRequests: ["request", "request2"]
      }
    }
  };
  const wrapper = shallow(<Dashboard {...PropsWithRequests} />);
  it("displays friend requests array", () => {
    expect(wrapper.find(FriendRequests).length).toEqual(1);
  });
});
