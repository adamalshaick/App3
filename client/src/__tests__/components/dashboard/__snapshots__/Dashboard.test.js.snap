// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render component renders component 1`] = `
<article
  className="container entry"
>
  <div
    className="row mt-5"
  >
    <section
      className="col-md-6"
      style={
        Object {
          "borderRight": "whitesmoke solid 2px",
        }
      }
    >
      <Connect(ProfileContent)
        profile={
          Object {
            "friendRequests": Array [],
          }
        }
      />
    </section>
    <section
      className="col-md-6"
    >
      <styled.header
        className="text-center"
      >
        Your Friends
      </styled.header>
      <hr />
      <Connect(ComposedComponent)
        currentProfile={
          Object {
            "friendRequests": Array [],
          }
        }
      />
    </section>
  </div>
</article>
`;
